opensrp-app/src/test/java/org/smartregister/service/HTTPAgentTest.java

```diff
@@ -560,4 +566,342 @@ public void testOauth2authenticateRefreshTokenInvokesOauth2authenticateCoreWithC
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(TEST_BASE_URL).when(dristhiConfiguration).dristhiBaseURL();
+        Mockito.doReturn(TEST_USERNAME).when(allSharedPreferences).fetchRegisteredANM();
+        Mockito.doReturn(httpURLConnection).when(httpAgentSpy).getHttpURLConnection("https://my-server.com/user-details?anm-id=" + TEST_USERNAME);
+
+        Mockito.doReturn(HttpURLConnection.HTTP_UNAUTHORIZED).when(httpURLConnection).getResponseCode();
+
+        boolean isVerified = httpAgentSpy.verifyAuthorization();
+        Assert.assertFalse(isVerified);
+
+    }
+
+    @Test
+    public void testUrlCanBeAccessWithGivenCredentialsReturnsUnauthorizedResponse() throws Exception {
+
+        PowerMockito.mockStatic(Base64.class);
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(TEST_BASE_URL).when(dristhiConfiguration).dristhiBaseURL();
+        Mockito.doReturn(TEST_USERNAME).when(allSharedPreferences).fetchRegisteredANM();
+        Mockito.doReturn(httpURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+
+        Mockito.doReturn(HttpURLConnection.HTTP_UNAUTHORIZED).when(httpURLConnection).getResponseCode();
+
+        LoginResponse response = httpAgentSpy.urlCanBeAccessWithGivenCredentials(USER_DETAILS_ENDPOINT, TEST_USERNAME, TEST_PASSWORD);
+        Assert.assertNotNull(response);
+        Assert.assertNotNull(response.message());
+        Assert.assertNull(response.payload());
+        Assert.assertEquals("Please check the credentials", response.message());
+
+    }
+
+    @Test
+    public void testUrlCanBeAccessWithGivenCredentialsReturnsErrorResponseForMalformedURL() throws Exception {
+
+        PowerMockito.mockStatic(Base64.class);
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(TEST_BASE_URL).when(dristhiConfiguration).dristhiBaseURL();
+        Mockito.doReturn(TEST_USERNAME).when(allSharedPreferences).fetchRegisteredANM();
+        Mockito.doReturn(httpURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+
+        Mockito.doThrow(new MalformedURLException()).when(httpURLConnection).getResponseCode();
+
+        LoginResponse response = httpAgentSpy.urlCanBeAccessWithGivenCredentials(USER_DETAILS_ENDPOINT, TEST_USERNAME, TEST_PASSWORD);
+        Assert.assertNotNull(response);
+        Assert.assertNull(response.payload());
+        Assert.assertNotNull(response.message());
+        Assert.assertEquals(LoginResponse.MALFORMED_URL.name(), response.name());
+
+    }
+
+    @Test
+    public void testUrlCanBeAccessWithGivenCredentialsReturnsCorrectErrorResponseForSocketTimeout() throws Exception {
```

Comment:
I think this test name is incorrect. We have a socket timeout being thrown on line 707 in this file. I am not sure if this test is a duplicate since we also have another test that checks for MalformedURLException

Reply:
Fixed